### PR TITLE
[SR-4934] StarRocks external table can have different name with source table

### PR DESCRIPTION
**Pull request by wuleistarrocks**

### CODE REVIEW

**Comment by wuleistarrocks** [comment]
Fix #546

File: fe/fe-core/src/main/java/com/starrocks/catalog/ExternalOlapTable.java [hunk 1]
```diff
@@ -153,24 +152,28 @@ public void parseFromProperties(Map<String, String> properties) throws DdlExcept
...
-    public ExternalOlapTable(long tableId, String tableName, List<Column> baseSchema, KeysType keysType,
+    public ExternalOlapTable(long sourceDb, long tableId, String tableName, List<Column> baseSchema, KeysType keysType,
                              PartitionInfo partitionInfo, DistributionInfo defaultDistributionInfo,
                              TableIndexes indexes, Map<String, String> properties)
         throws DdlException {
         super(tableId, tableName, baseSchema, keysType, partitionInfo, defaultDistributionInfo, indexes);
         setType(TableType.OLAP_EXTERNAL);
         dbId = -1;
+        sourceDbId = sourceDb;
+        LOG.info("new external table, dbid: {}", sourceDb);
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/catalog/ExternalOlapTable.java:176]: remove not used log

**Review by chaoyli — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/ExternalOlapTable.java [hunk 2]
```diff
@@ -153,24 +152,27 @@ public void parseFromProperties(Map<String, String> properties) throws DdlExcept
     }
 
     private long dbId;
+    private long sourceDbId;
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/catalog/ExternalOlapTable.java:155]: dbId and sourceDbId are confused, better to use sourceDbId as db id of remote table?
Why not put the remote db id in the ExternalTableInfo?

File: fe/fe-core/src/main/java/com/starrocks/catalog/ExternalOlapTable.java [hunk 3]
```diff
@@ -200,13 +206,15 @@ public void write(DataOutput out) throws IOException {
         Text.writeString(out, type.name());
         out.writeLong(id);
         Text.writeString(out, name);
+        out.writeLong(sourceDbId);
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/catalog/ExternalOlapTable.java:209]: Better to use json to persist, or you must increase the meta version if you add more persist member.

File: fe/fe-core/src/main/java/com/starrocks/external/starrocks/TableMetaSyncer.java [hunk 4]
```diff
@@ -34,7 +34,7 @@ public void syncTable(ExternalOlapTable table) {
 
         TGetTableMetaRequest request = new TGetTableMetaRequest();
         request.setDb_name(table.getExternalInfo().getDbName());
-        request.setTable_name(table.getName());
+        request.setTable_name(table.getExternalInfo().getTableName());
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/external/starrocks/TableMetaSyncer.java:37]: Better to use db id and table id, which can be compatible with remote db and table renaming.

Comment (wuleistarrocks) [reply fe/fe-core/src/main/java/com/starrocks/external/starrocks/TableMetaSyncer.java:37]: For the first time get table meta from remote cluster, we have nor idea of table id

File: fe/fe-core/src/main/java/com/starrocks/catalog/ExternalOlapTable.java (hunk 3 above)
Comment (wuleistarrocks) [reply fe/fe-core/src/main/java/com/starrocks/catalog/ExternalOlapTable.java]: ok

**Review by gengjun-git — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Pull request merged**